From 2faf8669a2ba20250ffe2d8a1b63d4f8ef8c5b74 Mon Sep 17 00:00:00 2001
From: Satish Balay <balay@mcs.anl.gov>
Date: Fri, 27 Feb 2015 20:44:31 -0600
Subject: [PATCH] fix duplicate symbols [mc64dd_,mc64ed_,mc64fd_] when linking
 with superlu

With mc64id_() -> mc64id_dist() change [no longer underscore function] this is not
 needed [or cannot be used from fortran] - so MC64ID_DIST etc fortran mappings can
be removed from Cnames.h - and have prototypes in public include files?

     CLINKER /home/balay/petsc/arch-linux2-c-debug/lib/libpetsc.so.3.05.3
/home/balay/petsc/arch-linux2-c-debug/lib/libsuperlu_4.3.a(mc64ad.o): In function `mc64dd_':
/home/balay/petsc/arch-linux2-c-debug/externalpackages/SuperLU_4.3/SRC/mc64ad.c:1086: multiple definition of `mc64dd_'
/home/balay/petsc/arch-linux2-c-debug/lib/libsuperlu_dist_4.0.a(mc64ad_dist.o):/home/balay/petsc/arch-linux2-c-debug/externalpackages/SuperLU_DIST_4.0/SRC/mc64ad_dist.c:1097: first defined here
/home/balay/petsc/arch-linux2-c-debug/lib/libsuperlu_4.3.a(mc64ad.o): In function `mc64ed_':
/home/balay/petsc/arch-linux2-c-debug/externalpackages/SuperLU_4.3/SRC/mc64ad.c:1164: multiple definition of `mc64ed_'
/home/balay/petsc/arch-linux2-c-debug/lib/libsuperlu_dist_4.0.a(mc64ad_dist.o):/home/balay/petsc/arch-linux2-c-debug/externalpackages/SuperLU_DIST_4.0/SRC/mc64ad_dist.c:1175: first defined here
/home/balay/petsc/arch-linux2-c-debug/lib/libsuperlu_4.3.a(mc64ad.o): In function `mc64fd_':
/home/balay/petsc/arch-linux2-c-debug/externalpackages/SuperLU_4.3/SRC/mc64ad.c:1259: multiple definition of `mc64fd_'
/home/balay/petsc/arch-linux2-c-debug/lib/libsuperlu_dist_4.0.a(mc64ad_dist.o):/home/balay/petsc/arch-linux2-c-debug/externalpackages/SuperLU_DIST_4.0/SRC/mc64ad_dist.c:1270: first defined here
---
 SRC/mc64ad_dist.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/SRC/mc64ad_dist.c b/SRC/mc64ad_dist.c
index 1610679..66a6ee0 100644
--- a/SRC/mc64ad_dist.c
+++ b/SRC/mc64ad_dist.c
@@ -15,6 +15,10 @@
 #define abs(x) ((x) >= 0 ? (x) : -(x))
 #define min(a,b) ((a) < (b)) ? (a) : (b)
 
+#define mc64dd_ mc64dd_dist
+#define mc64ed_ mc64ed_dist
+#define mc64fd_ mc64fd_dist
+
 /* Table of constant values */
 
 static int_t c__1 = 1;
-- 
2.7.4.1.g5468f9e

